DigitalLearningSolutions.Data.Tests/DataServices/UserDataServiceTests/DelegateUserDataServiceTests.cs

```diff
@@ -520,5 +521,70 @@ FROM DelegateAccounts
             ).SingleOrDefault();
             result.Should().Be(hash);
         }
+
+        [Test]
+        public void LinkDelegateAccountToNewUser_updates_UserId_and_sets_RegistrationConfirmationHash_to_null()
+        {
+            using var transaction = new TransactionScope();
+
+            // Given
+            const int centreId = 7;
+            const int newUserId = 2;
+
+            var userId = connection.QuerySingle<int>(
+                @"INSERT INTO Users
+                (
+                    PrimaryEmail,
+                    PasswordHash,
+                    FirstName,
+                    LastName,
+                    JobGroupID,
+                    Active,
+                    FailedLoginCount,
+                    HasBeenPromptedForPrn,
+                    HasDismissedLhLoginWarning
+                )
+                OUTPUT Inserted.ID
+                VALUES
+                ('LinkDelegateAccountToNewUser_updates_UserId_and_sets_RegistrationConfirmationHash_to_null@email.com', 'password', 'test', 'user', 1, 1, 0, 1, 1)"
+            );
+
+            var delegateId = connection.QuerySingle<int>(
+                @"INSERT INTO DelegateAccounts
+                (
+                    UserID,
+                    CentreID,
+                    DateRegistered,
+                    CandidateNumber,
+                    Approved,
+                    Active,
+                    ExternalReg,
+                    SelfReg,
+                    RegistrationConfirmationHash
+                )
+                OUTPUT Inserted.ID
+                VALUES
+                (@userId, @centreId, CURRENT_TIMESTAMP, 'candidate_number', 1, 1, 0, 1, 'hash')",
+                new { userId, centreId }
+            );
+
+            var oldUser = userDataService.GetUserAccountById(userId);
+            var newUser = userDataService.GetUserAccountById(newUserId);
+            var newUserDelegateAccounts = userDataService.GetDelegateAccountsByUserId(newUserId);
+
+            // When
+            userDataService.LinkDelegateAccountToNewUser(userId, newUserId, centreId);
+
+            // Then
+            oldUser.Should().NotBeNull();
+            newUser.Should().NotBeNull();
+
+            newUserDelegateAccounts.Should().NotContain(delegateAccount => delegateAccount.CentreId == centreId);

```

Comment:
The purpose of this is to confirm that the database has updated successfully. 
Since you can run an SQL UPDATE command on a record that does not exist, and this does not result in an exception, I wanted to prove that there were records to be updated before the update happened, and that they were not the same after the update happened.